packages/aws/src/lib/CostAndUsageReports.ts

```diff
@@ -156,58 +156,67 @@ export default class CostAndUsageReports {
-        null,
-        '',
-        '',
-        inputDataRow.region,
-        inputDataRow.serviceName,
-        inputDataRow.usageType,
-        inputDataRow.usageUnit,
-        inputDataRow.vCpus != '' ? parseFloat(inputDataRow.vCpus) : null,
-        1,
-        1,
-        {},
-      )
-
-      const footprintEstimate = this.getFootprintEstimateFromUsageRow(
-        costAndUsageReportRow,
-        unknownRows,
-      )
-
-      if (footprintEstimate) {
-        result.push({
-          serviceName: inputDataRow.serviceName,
-          region: inputDataRow.region,
-          usageType: inputDataRow.usageType,
-          vCpus: inputDataRow.vCpus,
-          kilowattHours: footprintEstimate.kilowattHours,
-          co2e: footprintEstimate.co2e,
-        })
+      try {
+          const costAndUsageReportRow = new CostAndUsageReportsRow(
+            null,
+            '',
+            '',
+            inputDataRow.region,
+            inputDataRow.serviceName,
+            inputDataRow.usageType,
+            inputDataRow.usageUnit,
+            inputDataRow.vCpus != '' ? parseFloat(inputDataRow.vCpus) : null,
+            1,
+            1,
+            {},
+          );
+
+          const footprintEstimate = this.getFootprintEstimateFromUsageRow(
+            costAndUsageReportRow,
+            unknownRows,
+          );
+
+          if (footprintEstimate) {
+            result.push({
+              serviceName: inputDataRow.serviceName,
+              region: inputDataRow.region,
+              usageType: inputDataRow.usageType,
+              vCpus: inputDataRow.vCpus,
+              kilowattHours: footprintEstimate.kilowattHours,
+              co2e: footprintEstimate.co2e,
+            });
+          }
+      } catch (error) {
+          console.error(`Error processing inputDataRow: ${JSON.stringify(inputDataRow)}. Error: ${error.message}`);
       }
-    })
+    });
 
     if (result.length > 0) {
       unknownRows.map((inputDataRow: CostAndUsageReportsRow) => {
-        const footprintEstimate = this.getEstimateForUnknownUsage(inputDataRow)
-        if (footprintEstimate)
-          result.push({
-            serviceName: inputDataRow.serviceName,
-            region: inputDataRow.region,
-            usageType: inputDataRow.usageType,
-            vCpus: inputDataRow.vCpus,
-            kilowattHours: footprintEstimate.kilowattHours,
-            co2e: footprintEstimate.co2e,
-          })
-      })
+        try {
+            const footprintEstimate = this.getEstimateForUnknownUsage(inputDataRow);
+            if (footprintEstimate)
+              result.push({
+                serviceName: inputDataRow.serviceName,
+                region: inputDataRow.region,
+                usageType: inputDataRow.usageType,
+                vCpus: inputDataRow.vCpus,
+                kilowattHours: footprintEstimate.kilowattHours,
+                co2e: footprintEstimate.co2e,
+              });
+        } catch (error) {
```

Comment:
Same thought as above. I think if the error we are trying to fix is the NaN issue, we may be better to error handle at the root where the arithmetic occurs to aggregate the results. I could also see this potentially being console warnings.